troubadix/plugins/duplicate_oid.py

```diff
@@ -54,43 +55,46 @@ def run(
         invalids = []
 
         for nasl_file in nasl_files:
-            if nasl_file.suffix == ".nasl":
-                oid = None
-                file_name = get_path_from_root(nasl_file)
-                content = nasl_file.read_text(encoding="latin-1")
-                # search for deprecated script_id
+            if not nasl_file.suffix == ".nasl":
+                continue
+
+            oid = None
+            file_name = get_path_from_root(nasl_file)
+            content = nasl_file.read_text(encoding=CURRENT_ENCODING)
+            # search for deprecated script_id
+            match = get_special_script_tag_pattern(SpecialScriptTag.OID).search(
+                content
+            )
+
+            if match:
+                oid = match.group("oid")
+            else:
                 match = get_special_script_tag_pattern(
-                    SpecialScriptTag.OID
```

Comment:
why change it? but not use later the correct mach?

```
oid = OPENVAS_OID_PREFIX + match.group("id")
```

Reply:
I think it is already a typo that there was used `oid` in the old code.

Reply:
I didn't change anything regarding the content of this code. Just the if clause to remove the else branch.